[DNM] Add FallbackBannerConductor

Abban opened the pull request:
This banner conductor will show a fallback banner if the width is below a threshold or the main banner has a size issue.

Ticket: https://phabricator.wikimedia.org/T351323

gbirke commented on wikipedia.de/banners/Banner.jsx:
```diff
@@ -1,28 +1,32 @@
-import {Component, h, Fragment } from 'preact';
+// eslint-disable-next-line no-unused-vars
+import { Component, h, Fragment } from 'preact';
+// eslint-disable-next-line no-unused-vars
```
I think the usage of imported components in JSX should be configured in our rules with [jsx-uses-vars](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-uses-vars.md), which is already in the `eslint-plugin-react` but only needs to be configured differently.

timEulitz replied:
I have tried that but whenever I add `"jsx-uses-vars": 1` to the rules, ESLint seems to bug out and stops recognizing ` eslint-disable-next-line no-unused-vars` altogether even though we still need it. If you'd like, you can double-check and see if you can find a solution but I am bit out of ideas at this point 🤔 .

Abban committed:
Add FallbackBannerConductor

This banner conductor will show a fallback banner if the
width is below a threshold or the main banner has a size
issue.

Ticket: https://phabricator.wikimedia.org/T351323

Abban commented on banners/desktop/banner_ctrl.ts:
```diff
@@ -45,6 +46,8 @@ const app = createVueApp( BannerConductor, {
 	resizeHandler: new WindowResizeHandler(),
 	banner: Banner,
+	fallbackBanner: FallbackBanner,
+	minWidthForMainBanner: 800,
```
This could probably go into the banner config. Actually we probably should be initialisiing that early in the entry point then using values from it so there's no repetition and the values ae more descriptive when passed into the various constructors and props.

gbirke replied:
Sorry, I don't understand this comment - which "banner config" do you mean? `campaign_info.toml`? I'm not sure if we should put too much in there, I think the entry point is a better place to put such "configuration".

Abban replied:
No, there's a bannerConfig prop. It also made me think that we could be defining that separately in the entry points eg:

```js
const bannerConfig = {
    minAllowedWidth: 400,
    minWidthForMainBanner: 800,
    bannerDelay: 7500,
    transitionDuration: 1000
};
```

It'd then put those values together and make it a bit more explicit when they're used eg:

```js
new WindowSizeIssueChecker( 400 )
```

would be 

```js
new WindowSizeIssueChecker( bannerConfig.minAllowedWidth )
```

gbirke replied:
That makes perfect sense and is a good idea, thanks!

gbirke reviewed (approved):
I've user-tested this and looked at the code and I'm happy with it.

Abban commented:
Closing this as it was merged as part of https://github.com/wmde/fundraising-banners/pull/289
